### PR TITLE
Fix unhandled exception when a colorfill plot's scale is set to logarithmic and negative limits are entered

### LINKED ISSUE #28226: Colorfill Plot throws Exception changing to log scale
### Expected behavior
- Should be able to change the colour scale to log
### Actual behavior
- Throws exception
```
  File "/opt/mantidnightly/lib/workbench/plotting/figureinteraction.py", line 743, in _change_colorbar_axes
    datafunctions.update_colorbar_scale(self.canvas.figure, image, scale_type, image.norm.vmin, image.norm.vmax)
  File "/opt/mantidnightly/lib/mantid/plots/datafunctions.py", line 1024, in update_colorbar_scale
    if locator.tick_values(vmin=vmin, vmax=vmax).size == 0:
  File "/usr/lib/python3/dist-packages/matplotlib/ticker.py", line 2123, in tick_values
    "Data has no positive values, and therefore can not be "
ValueError: Data has no positive values, and therefore can not be log-scaled.
```

### Steps to reproduce the behavior
Run the following:
```
Load(Filename='GEM38370_Focussed.nxs', OutputWorkspace='GEM38370_Focussed')
ConvertUnits(InputWorkspace='GEM38370_Focussed', OutputWorkspace='ws_lambda', Target='Wavelength')
```
- Plot -> Colorfill
- Right click figure -> Colour Bar -> Log -> Stops working

### PR DESCRIPTION
**Pull request by PhilColebrooke**
**To test:**
Load a workspace, right-click it and select Plot->Colorfill.
Open figure options, go to Images tab.
Check that when the scale is set to Linear you can enter negative numbers for min value and max value.
Check that when the scale is set to Logarithmic any negative values become zero and you can't enter negative values.
Apply the settings with the scale set to Logarithmic and close figure options.
Double-click the colorbar axis, check that no exception is thrown if you enter a negative number for the min and max value. The min value should change to 1e-6 and the max value should change to 1.

Fixes #28226 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by alicerussell1 — CHANGES_REQUESTED** [review]
As discussed it should display a clear warning that they have entered an invalid number and it has been changed. It is also possible to enter negative numbers in the figure options if you have changed the values in the double-click box first.

**Review by alicerussell1 — DISMISSED** [review]
This looks good to me 👍

**Comment by PhilColebrooke** [comment]
Realised I changed an attribute name for seemingly no reason so I've changed it back... 🤔

**Review by alicerussell1 — APPROVED** [review]
Still works 👍

File: qt/applications/workbench/workbench/plotting/propertiesdialog.py [hunk 1]
```diff
@@ -203,11 +203,19 @@ def __init__(self, canvas, axes):
         self.create_model()
 
     def changes_accepted(self):
-        super(ColorbarAxisEditor, self).changes_accepted()
```
Comment (martyngigg) [inline qt/applications/workbench/workbench/plotting/propertiesdialog.py]: Can I just check that this call to the super class was intended to be removed?

Comment (PhilColebrooke) [reply qt/applications/workbench/workbench/plotting/propertiesdialog.py]: Yeah so the call to the super class caused problems is if the `ValueError` below was raised.  I also realised that there is some unnecessary stuff in the super class function when dealing with the colorbar axis editor so I've just moved the things that are needed into this function.

Comment (PhilColebrooke) [reply qt/applications/workbench/workbench/plotting/propertiesdialog.py]: > Can I just check that this call to the super class was intended to be removed?

Yeah so the call to the super class caused problems if the ValueError below was raised. I also realised that there is some unnecessary stuff in the super class function when dealing with the colorbar axis editor so I've just moved the things that are needed into this function.

Comment (martyngigg) [reply qt/applications/workbench/workbench/plotting/propertiesdialog.py]: Thanks for clarifying. :+1:

**Pull request merged**